diff --unified --recursive --text --color st/config.def.h st.new/config.def.h
--- st/config.def.h	2020-06-18 20:06:17.786653639 +0200
+++ st.new/config.def.h	2020-06-18 20:14:13.469986441 +0200
@@ -1,3 +1,4 @@
+char *externalpipe_sigusr1[] = {"/bin/sh", "-c", "externalpipe_buffer.sh st_strings_read"};
 /* See LICENSE file for copyright and license details. */
 
 /*
diff --unified --recursive --text --color st/st.c st.new/st.c
--- st/st.c	2020-06-18 20:06:17.789986972 +0200
+++ st.new/st.c	2020-06-18 20:14:13.473319775 +0200
@@ -163,6 +163,7 @@
 static void execsh(char *, char **);
 static void stty(char **);
 static void sigchld(int);
+static void sigusr1(int);
 static void ttywriteraw(const char *, size_t);
 
 static void csidump(void);
@@ -739,6 +740,13 @@
 }
 
 void
+sigusr1(int unused)
+{
+  static Arg a = {.v = externalpipe_sigusr1};
+  externalpipe(&a);
+}
+
+void
 sigchld(int a)
 {
 	int stat;
@@ -784,6 +792,12 @@
 int
 ttynew(char *line, char *cmd, char *out, char **args)
 {
+	static struct sigaction sa;
+	sa.sa_handler = sigusr1;
+	sigemptyset(&sa.sa_mask);
+	sa.sa_flags = SA_RESTART;
+	sigaction(SIGUSR1, &sa, NULL);
+
 	int m, s;
 
 	if (out) {
diff --unified --recursive --text --color st/st.h st.new/st.h
--- st/st.h	2020-06-18 20:06:17.786653639 +0200
+++ st.new/st.h	2020-06-18 20:14:35.149986423 +0200
@@ -126,6 +126,7 @@
 #endif
 
 /* config.h globals */
+extern char *externalpipe_sigusr1[];
 extern char *utmp;
 extern char *scroll;
 extern char *stty_args;
