diff --color -up st/config.def.h st.new/config.def.h
--- st/config.def.h	2021-02-23 16:15:05.719363612 +0100
+++ st.new/config.def.h	2021-02-23 16:16:28.599361904 +0100
@@ -232,6 +232,7 @@ static Shortcut shortcuts[] = {
 	{ ShiftMask,            XK_Insert,      clippaste,      {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
 	{ MODKEY,               XK_l,           copyurl,        {.i =  0} },
+	{ MODKEY,               XK_o,           opencopied,     {.v = "xdg-open"} },
 	{ ShiftMask,            XK_Page_Up,     kscrollup,      {.i = -1} },
 	{ ShiftMask,            XK_Page_Down,   kscrolldown,    {.i = -1} },
 };
Common subdirectories: st/.git and st.new/.git
diff --color -up st/st.h st.new/st.h
--- st/st.h	2021-02-23 16:15:05.719363612 +0100
+++ st.new/st.h	2021-02-23 16:16:50.349361455 +0100
@@ -86,6 +86,7 @@ void externalpipe(const Arg *);
 void kscrolldown(const Arg *);
 void kscrollup(const Arg *);
 void newterm(const Arg *);
+void opencopied(const Arg *);
 void printscreen(const Arg *);
 void printsel(const Arg *);
 void sendbreak(const Arg *);
diff --color -up st/x.c st.new/x.c
--- st/x.c	2021-02-23 16:15:05.719363612 +0100
+++ st.new/x.c	2021-02-23 16:20:21.909357094 +0100
@@ -507,6 +507,29 @@ bpress(XEvent *e)
 }
 
 void
+opencopied(const Arg *arg)
+{
+	const size_t max_cmd = 2048;
+	const char *clip = xsel.clipboard;
+	if(!clip) {
+		fprintf(stderr, "Warning: nothing copied to clipboard\n");
+		return;
+	}
+
+	/* account for space/quote (3) and \0 (1) and & (1) */
+	char cmd[max_cmd + strlen(clip) + 5];
+	strncpy(cmd, (char *)arg->v, max_cmd);
+	cmd[max_cmd] = '\0';
+
+	strcat(cmd, " \"");
+	strcat(cmd, clip);
+	strcat(cmd, "\"");
+	strcat(cmd, "&");
+
+	system(cmd);
+}
+
+void
 propnotify(XEvent *e)
 {
 	XPropertyEvent *xpev;
